app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -20,6 +20,10 @@ class PuppetclassLookupKey < LookupKey
 
   scope :smart_class_parameters, -> { joins(:environment_classes).readonly(false) }
 
+  def can_edit?
```

Comment:
This name is a bit confusing - the LK isn't doing the editing. Perhaps change it to `editable_by_user?` or something similar?